### DISCUSSION TITLE
Welcome to Rotten-Scripts Discussions!

### DISCUSSION THREAD

**Discussion by HarshCasper** [HarshCasper/Rotten-Scripts#584] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by vybhav72954** [comment]
Hello everyone, Vybhav this side. Some of you might have seen me in comments here and there on multiple issues and PRs. I just hang out o GitHub I guess. _LOL_

**Comment by Vishal-raj-1** [comment]
Hello Everyone, I am Vishal, an open source enthusiast. I am participating in KWOC and NWOC both to learn new skills and work on my frontend skills. I am here to learn with you all guys.

**Comment by vybhav72954** [comment]
@HarshCasper The code for #318 is actually simple, I can do that, but the problem is am really busy right now, (1 more day and my internship issue are over), so yes am free from 31st December, I can start working on this issue and actually every issue that has gone stale because of me from tomorrow. You can assign this to me maybe, but I will work do anything from the day after tomorrow at the earliest.

**Reply by HarshCasper** [reply]
No worries @vybhav72954 

You can take your time 😄

**Comment by vybhav72954** [comment]
@HarshCasper Can you possibly make a Project/Branch "Project Preferred" so that I can start working on the Website, (I know we decided to work during `SLoP` am sorry I was unable to)

**Reply by vybhav72954** [reply]
> @HarshCasper Can you possibly make a Project/Branch "Project Preferred" so that I can start working on the Website, (I know we decided to work during `SLoP` am sorry I was unable to)

This can then also be used to divide the Project flowchart like you can open a few issues from your side in every language, (Right Now Python dominates everything) which is not a bad thing, but maybe we can open more issues in other language and with suitable labels, more people will be attracted.

Similarly, we can have a different board just for code styling and correction, where weekly you can link 5-10 directories and a person can style them and their respective README, using the actions that we already have.

This is not a clear-cut solution and considering you have a better experience than me, I believe you can judge if this is useful or not.

**Reply by HarshCasper** [reply]
Hi @vybhav72954 

Right now, we are discussing on the Website here: https://github.com/HarshCasper/Rotten-Scripts/issues/656 

Since I am completely noob around this, I have let @avinal take care of the same. Would you like to work on a full-fledged site probably that would cover docs and get started content more extensively?

**Reply by HarshCasper** [reply]
> > @HarshCasper Can you possibly make a Project/Branch "Project Preferred" so that I can start working on the Website, (I know we decided to work during `SLoP` am sorry I was unable to)
> 
> This can then also be used to divide the Project flowchart like you can open a few issues from your side in every language, (Right Now Python dominates everything) which is not a bad thing, but maybe we can open more issues in other language and with suitable labels, more people will be attracted.
> 
> Similarly, we can have a different board just for code styling and correction, where weekly you can link 5-10 directories and a person can style them and their respective README, using the actions that we already have.
> 
> This is not a clear-cut solution and considering you have a better experience than me, I believe you can judge if this is useful or not.

Quite a sound idea. Once @CharvyJain finishes the work on updating the README, we will try to ensure that all the Scripts are broken into sub-topics and new contributors can now work on enhancing them up using DeepSource and your CI config, to improve the Code Quality a bit.

**Reply by vybhav72954** [reply]
> Hi @vybhav72954
> 
> Right now, we are discussing on the Website here: #656
> 
> Since I am completely noob around this, I have let @avinal take care of the same. Would you like to work on a full-fledged site probably that would cover docs and get started content more extensively?

Static Websites are not my thing honestly, Although I can definitely start preparing a full-fledged website based on React (Native). Just allow me to get done with my stale issues, after that I will send you a small proposal/architecture (via email maybe).

After having a small discussion about that, we can open that to everyone here, so that after some constructive feedback, by the end of these WoC, we can have a running website.

Meanwhile, @avinal I can help you with static (though i have only made a half static website once 😛 )  as well in case you need some 😄

**Reply by vybhav72954** [reply]
> > > @HarshCasper Can you possibly make a Project/Branch "Project Preferred" so that I can start working on the Website, (I know we decided to work during `SLoP` am sorry I was unable to)
> > 
> > 
> > This can then also be used to divide the Project flowchart like you can open a few issues from your side in every language, (Right Now Python dominates everything) which is not a bad thing, but maybe we can open more issues in other language and with suitable labels, more people will be attracted.
> > Similarly, we can have a different board just for code styling and correction, where weekly you can link 5-10 directories and a person can style them and their respective README, using the actions that we already have.
> > This is not a clear-cut solution and considering you have a better experience than me, I believe you can judge if this is useful or not.
> 
> Quite a sound idea. Once @CharvyJain finishes the work on updating the README, we will try to ensure that all the Scripts are broken into sub-topics and new contributors can now work on enhancing them up using DeepSource and your CI config, to improve the Code Quality a bit.

I believe making a GitHub Project would be the Key here

**Reply by HarshCasper** [reply]
> > > > @HarshCasper Can you possibly make a Project/Branch "Project Preferred" so that I can start working on the Website, (I know we decided to work during `SLoP` am sorry I was unable to)
> > > 
> > > 
> > > This can then also be used to divide the Project flowchart like you can open a few issues from your side in every language, (Right Now Python dominates everything) which is not a bad thing, but maybe we can open more issues in other language and with suitable labels, more people will be attracted.
> > > Similarly, we can have a different board just for code styling and correction, where weekly you can link 5-10 directories and a person can style them and their respective README, using the actions that we already have.
> > > This is not a clear-cut solution and considering you have a better experience than me, I believe you can judge if this is useful or not.
> > 
> > 
> > Quite a sound idea. Once @CharvyJain finishes the work on updating the README, we will try to ensure that all the Scripts are broken into sub-topics and new contributors can now work on enhancing them up using DeepSource and your CI config, to improve the Code Quality a bit.
> 
> I believe making a GitHub Project would be the Key here

Yeah. Something that would ensure that we remove duplicated code and bad code altogether. I will get working on it by the end of this week.

**Reply by HarshCasper** [reply]
> > Hi @vybhav72954
> > Right now, we are discussing on the Website here: #656
> > Since I am completely noob around this, I have let @avinal take care of the same. Would you like to work on a full-fledged site probably that would cover docs and get started content more extensively?
> 
> Static Websites are not my thing honestly, Although I can definitely start preparing a full-fledged website based on React (Native). Just allow me to get done with my stale issues, after that I will send you a small proposal/architecture (via email maybe).
> 
> After having a small discussion about that, we can open that to everyone here, so that after some constructive feedback, by the end of these WoC, we can have a running website.
> 
> Meanwhile, @avinal I can help you with static (though i have only made a half static website once 😛 ) as well in case you need some 😄

Sounds awesome. We need a minimal Tech Stack for this. I will possibly discuss the features with you on another Discussions Thread about the same, where everyone can participate.

**Reply by avinal** [reply]
> 
> 
> Hi @vybhav72954
> 
> Right now, we are discussing on the Website here: #656
> 
> Since I am completely noob around this, I have let @avinal take care of the same. Would you like to work on a full-fledged site probably that would cover docs and get started content more extensively?

If someone is ready for a full fledged website then we should do that, because SSGs can only work within a template, so, has limited capabilities. But also its much easier to implement and maintain. Let me know the final decision because I am really a noob in web development 😂

**Reply by HarshCasper** [reply]
> > Hi @vybhav72954
> > Right now, we are discussing on the Website here: #656
> > Since I am completely noob around this, I have let @avinal take care of the same. Would you like to work on a full-fledged site probably that would cover docs and get started content more extensively?
> 
> If someone is ready for a full fledged website then we should do that, because SSGs can only work within a template, so, has limited capabilities. But also its much easier to implement and maintain. Let me know the final decision because I am really a noob in web development 😂

Let's have a SSG for now. I am actually looking for a Site so that we can attract some new Contributors.

**Reply by vybhav72954** [reply]
Yes, because the setup of a static website is more straightforward, it's easier, we can have one quickly.

**Comment by avinal** [comment]
Hello @HarshCasper, it seems Auto Assign PR is not working, I have checked their repo, it seems there is an unsolved issue there. Should we find some alternatives?

**Comment by vybhav72954** [comment]
@avinal Do you mean Auto Assign Issues?

**Reply by avinal** [reply]
No, Auto assign Pull Request

**Reply by vybhav72954** [reply]
> No, Auto assign Pull Request

Ok I got it, The error says `Resource not accessible by integration`. This basically means that the SHA of PR is unable to access the GitHub credentials. This a usual problem with GitHub actions. But I think we can make it work.

I am pretty much sure, you said that in one of the PR, this action worked just the way it was supposed to, can you mention the PR here.

@HarshCasper I suggest you along with @avinal try to find an alternative, meanwhile, I will help @avinal in solving this as well, considering he made a useful action anyway.

**Reply by vybhav72954** [reply]
A quick workaround (read hack), is using the `RW_TOKEN`. But I doubt it can help in his case, as Forked Repos only have permission to READ.

**Comment by RutvikJ77** [comment]
Hey, @HarshCasper just pitching out, What if we create a package for these amazing scripts?

**Reply by HarshCasper** [reply]
It would be great @RutvikJ77 

If you have a proposal or would like to work with some core contributors please feel free to create a new discussion thread. 

Tagging @vybhav72954 and @seema1711 for the same

**Reply by vybhav72954** [reply]
@RutvikJ77 We can discuss this. Can you Start a discussion thread like Harsh mentioned

**Reply by RutvikJ77** [reply]
Sure!

**Reply by iseemasaharan** [reply]
Sounds good to me.

**Comment by haripriya9647** [comment]
Hey @vybhav72954 I have created pull request for issue #805 but 4 checks failed . Can you let me know how to resolve these.

**Reply by vybhav72954** [reply]
> Hey @vybhav72954 I have created pull request for issue #805 but 4 checks failed . Can you let me know how to resolve these.

I will check it, don't worry about the failing checks, I will help you with the PR

**Comment by mohanm18** [comment]
Hey @HarshCasper there's a folder name spelling mistake at (https://github.com/HarshCasper/Rotten-Scripts/tree/master/Python/Facts_Deskop_Notifier).
I guess it should be Facts_Desktop_Notifier. I was trying to raise the issue but there wasn't any form regarding spelling error 😅.

**Reply by HarshCasper** [reply]
Hi @mohanm18 — Please raise a PR for it directly 👍

**Reply by mohanm18** [reply]
Ok done 👍
